zerver/views/streams.py

```diff
@@ -754,6 +754,16 @@ def get_streams_backend(
     return json_success(request, data={"streams": streams})
 
 
+@has_request_variables
+def get_stream_backend(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    stream_id: int,
+) -> HttpResponse:
+    (stream, sub) = access_stream_by_id(user_profile, stream_id, allow_realm_admin=True)
+    return json_success(request, data={"stream": stream.to_dict()})
```

Comment:
Yeah, this looks correct. 

I think the ideal thing we'd do is have `Stream.to_dict()` and `Stream.get_client_data` return a type (and a list of that type) so that the relationship between their return format is enforced by more than convention. @PIG208 FYI as this may be a good little project for you as part of removing `Any` usage.

Reply:
Created #22123 for this @timabbott